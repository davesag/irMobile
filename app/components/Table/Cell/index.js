import React from 'react'
import PropTypes from 'prop-types'
import { Text } from 'react-native-paper'

import styles from '../styles'

const Cell = ({ value, numeric, style }) =>
  numeric ? (
    <Text style={[styles.data, style]}>{value}</Text>
  ) : (
    <Text style={style}>{value}</Text>
  )

Cell.propTypes = {
  value: PropTypes.string.isRequired,
  numeric: PropTypes.bool,
  style: PropTypes.object
}

Cell.defaultProps = {
  numeric: false,
  style: {}
}

export default Cell
