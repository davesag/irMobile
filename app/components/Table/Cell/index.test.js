import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import Cell from '.'

const value = 'some value'

let tree

describe('rendering', () => {
  describe('non numeric and no style', () => {
    beforeAll(() => {
      tree = renderer.create(<Cell value={value} />)
    })

    it('rendered correctly ', () => {
      expect(tree).toMatchSnapshot()
    })
  })

  describe('numeric and with style', () => {
    const style = { borderTop: 10 }

    beforeAll(() => {
      tree = renderer.create(<Cell value={value} style={style} numeric />)
    })

    it('rendered correctly ', () => {
      expect(tree).toMatchSnapshot()
    })
  })
})
