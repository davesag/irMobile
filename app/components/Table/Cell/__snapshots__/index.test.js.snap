// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering non numeric and no style rendered correctly  1`] = `null`;

exports[`rendering numeric and with style rendered correctly  1`] = `null`;
